tests/ga/test_update.py

```diff
@@ -1479,6 +1486,118 @@ def test_telemetry_heartbeat_creates_event(self, patch_add_event, patch_info, *_
         self.assertTrue(any(call_args[0] == "[HEARTBEAT] Agent {0} is running as the goal state agent {1}"
                             for call_args in patch_info.call_args), "The heartbeat was not written to the agent's log")
 
+    @contextlib.contextmanager
+    def _get_update_handler(self, iterations=1, test_data=DATA_FILE):
```

Comment:
Even though I modified an existing test class for these tests, I didnt change any of the existing tests because most of them are calling/mocking specific functions of the UpdateHandler and have the tests build around it (not to mention TestUpdate class has a huge amount of tests in it). The time investment would be huge to refactor these tests because I would have to fist understand what the test is doing and then modify the tests accordingly. In the end I chose not to do that. If you guys feel it should be done as part of this PR, I can start working on it.